symforce/codegen/codegen_util.py

```diff
@@ -340,18 +340,16 @@ def get_formatted_list(
             formatted_symbols = [sf.Symbol(key)]
             flattened_value = [value]
         elif issubclass(arg_cls, sf.Matrix):
-            if config.matrix_is_1d:
-                # TODO(nathan): Not sure this works for 2D matrices. Get rid of this.
-                formatted_symbols = [sf.Symbol(f"{key}[{j}]") for j in range(storage_dim)]
-            else:
-                # NOTE(brad): The order of the symbols must match the storage order of sf.Matrix
-                # (as returned by sf.Matrix.to_storage). Hence, if there storage order were
-                # changed to, say, row major, the below for loops would have to be swapped to
-                # reflect that.
-                formatted_symbols = []
-                for j in range(value.shape[1]):
-                    for i in range(value.shape[0]):
-                        formatted_symbols.append(sf.Symbol(f"{key}({i}, {j})"))
+            # NOTE(brad): The order of the symbols must match the storage order of sf.Matrix
+            # (as returned by sf.Matrix.to_storage). Hence, if there storage order were
+            # changed to, say, row major, the below for loops would have to be swapped to
+            # reflect that.
+            formatted_symbols = []
+            for j in range(value.shape[1]):
+                for i in range(value.shape[0]):
+                    formatted_symbols.append(
+                        sf.Symbol(config.format_matrix_accessor(key, i, j, shape=value.shape))
+                    )
```

Comment:
I didn't like how language specific logic (whether we render column vectors with one index (like `v[3]`) or with two (like `v(3, 0)`)) was leaked into this codegen logic, and wanted to pull that into the logic of `config.format_matrix_accessor`.

Also, if for some language config we wanted to render 2d matrices as 1 long array (so `m[r*COLS + c]` instead of `m[r, c]`, we'd wouldn't have enough information to do so.

So, by providing the shape of the matrix, `format_matrix_accessor` would have the info required to format correctly (regardless of row or column major storage order, and whether we want to store with 1 index or 2).

I chose to make `shape` a keyword-only argument to reduce the change of confusing the shape with the indices. As a side bonus, if we were to change the accessor to allow more than 2 indices, it should be easier this way.